movie-browser/src/App.js

```diff
@@ -1,31 +1,50 @@
 /* eslint-disable no-unused-vars */
 import React from 'react'
-import CardContainer from './CardContainer'
+import CardContainer from './view/CardContainer'
+import FeaturedMovie from './view/FeaturedMovie'
 import ApiReader from './model/ApiReader';
 
 // *** constant(s)
 const url = 'https://ghibliapi.herokuapp.com/films'
 
+// TODO: makes button working
+
 // *** Actual App
 class App extends React.Component {
-  state = {movies: []};
+  state = {movies: [], movie: {}, btnTxt: "View all"};
 
   // *** running ApiReader when mounted
   componentDidMount() {
     this.fetchData();
   }
 
+  handleClickButton = () => {
+    if (this.state.btnTxt === "View all") {
+      this.setState({
+        btnTxt: "Hide all"
+      })
+    } else {
+      this.setState({
+        btnTxt: "View all"
+      })
+    }
+  }
+
   async fetchData() {
-    const data = await ApiReader(url);
-    this.setState({ movies: data });
+    const data = await ApiReader(url)
+    
+    this.setState({ movies: data.moviesData });
```

Comment:
for every `setState` the component will re-render and so will its children. Performance wise we want to avoid as much re-renders as possible. So, I suggest that `featuredMovieData` should be another key in the state, this way u can just use `setState` once and set it all at once.